node_helper.js

```diff
@@ -20,92 +18,77 @@ module.exports = NodeHelper.create(
             console.log("Refresh every: " + refreshTime + "ms.")
 
             //Check whether unavailable leagues should be shown.
-            if (showUnavailable)
-            {
-                for (var j = 0; j < leagues.length; j++) 
-                {
+            if (showUnavailable) {
+                for (var j = 0; j < leagues.length; j++) {
                     var league = leagues[j];
-
+                    
                     var options = {
                         method: 'GET',
-                        url: 'http://api.football-data.org/v2/competitions/' + league.toString(),
                         headers: {
                             'X-Auth-Token': apiKey
-                        }
-                    };
+                    }
 
-                    request(options,
-                        function (error, response, body)
-                        {
+                    fetch('http://api.football-data.org/v2/competitions/' + league.toString(), {
+                          options
+                    }).then(function (error, response, body) {
```

Comment:
While it doesn't matter, you should use the .catch of promises instead of try catches inside the then.

Reply:
true